lib/throng.js

```diff
@@ -52,5 +60,32 @@ module.exports = function(startFn, options) {
       cluster.workers[id].kill();
     }
   }
-};
 
+  function monitor() {
+    function verifyMemory(worker, pid) {
+      pidusage.stat(pid, function(err, stat) {
+        if (err) return;
+
+        var memoryUsage = Math.floor(stat.memory);
+
+        if (memoryUsage > options.memoryLimit) {
+          console.log('killing process with pid ' + pid + ' for excessive memory usage')
+          worker.process.kill();
```

Comment:
Should we also try to force kill here after the grace period or killing the worker itself prevents restarts?